plugins/vpc-shared-eni/config/netconfig.go

```diff
@@ -30,33 +30,39 @@ import (
 // NetConfig defines the network configuration for the vpc-shared-eni plugin.
 type NetConfig struct {
 	cniTypes.NetConf
-	ENIName          string
-	ENIMACAddress    net.HardwareAddr
-	ENIIPAddress     *net.IPNet
-	VPCCIDRs         []net.IPNet
-	BridgeType       string
-	BridgeNetNSPath  string
-	IPAddress        *net.IPNet
-	GatewayIPAddress net.IP
-	InterfaceType    string
-	TapUserID        int
-	Kubernetes       KubernetesConfig
+	ENIName            string
+	ENIMACAddress      net.HardwareAddr
+	ENIIPAddress       *net.IPNet
+	ENIIPv6Address     *net.IPNet
```

Comment:
Instead of adding new fields, we should make the existing ENIIPAddress field an array that accepts both IPv4 and IPv6 addresses. See https://github.com/aws/amazon-vpc-cni-plugins/commit/a21d3a41f922e14c19387713df66be3e4ee1e1f6 for an example.

Reply:
There were a few differences in the way we configure ipv4 vs ipv6 so i had initially kept these separate. But i do see that if we have more addresses it makes more sense to have a list. Updating in the new PR.